Lib9c/Model/Item/ShopItem.cs

```diff
@@ -82,16 +82,32 @@ INonFungibleItem nonFungibleItem
 
         public ShopItem(Dictionary serialized)
         {
-            SellerAgentAddress = serialized["sellerAgentAddress"].ToAddress();
-            SellerAvatarAddress = serialized["sellerAvatarAddress"].ToAddress();
-            ProductId = serialized["productId"].ToGuid();
-            Price = serialized["price"].ToFungibleAssetValue();
-            ItemUsable = serialized.ContainsKey("itemUsable")
-                ? (ItemUsable) ItemFactory.Deserialize((Dictionary) serialized["itemUsable"])
-                : null;
-            Costume = serialized.ContainsKey("costume")
-                ? (Costume) ItemFactory.Deserialize((Dictionary) serialized["costume"])
-                : null;
+            if (serialized.ContainsKey(LegacySellerAgentAddressKey))
+            {
+                SellerAgentAddress = serialized[LegacySellerAgentAddressKey].ToAddress();
+                SellerAvatarAddress = serialized[LegacySellerAvatarAddressKey].ToAddress();
+                ProductId = serialized[LegacyProductIdKey].ToGuid();
+                Price = serialized[LegacyPriceKey].ToFungibleAssetValue();
+                ItemUsable = serialized.ContainsKey(LegacyItemUsableKey)
+                    ? (ItemUsable) ItemFactory.Deserialize((Dictionary) serialized[LegacyItemUsableKey])
+                    : null;
+                Costume = serialized.ContainsKey(LegacyCostumeKey)
+                    ? (Costume) ItemFactory.Deserialize((Dictionary) serialized[LegacyCostumeKey])
+                    : null;
+            }
+            else
+            {
+                SellerAgentAddress = serialized[SellerAgentAddressKey].ToAddress();
+                SellerAvatarAddress = serialized[SellerAvatarAddressKey].ToAddress();
+                ProductId = serialized[ProductIdKey].ToGuid();
+                Price = serialized[PriceKey].ToFungibleAssetValue();
+                ItemUsable = serialized.ContainsKey(ItemUsableKey)
+                    ? (ItemUsable) ItemFactory.Deserialize((Dictionary) serialized[ItemUsableKey])
+                    : null;
+                Costume = serialized.ContainsKey(CostumeKey)
+                    ? (Costume) ItemFactory.Deserialize((Dictionary) serialized[CostumeKey])
+                    : null;
+            }
```

Comment:
Like other deserialization does, how about removing the duplicated logic? First choose key by `useLegacy` and deserialize them.